pkg/nfs/nfs.go

```diff
@@ -32,17 +35,83 @@ type nfsDriver struct {
 	ns    *nodeServer
 	cap   []*csi.VolumeCapability_AccessMode
 	cscap []*csi.ControllerServiceCapability
+
+	csPlugin interface{}
 }
 
 const (
-	driverName = "csi-nfsplugin"
+	driverName       = "csi-nfsplugin"
+	pluginSymbolName = "NfsPlugin"
 )
 
 var (
 	version = "1.0.0-rc2"
 )
 
-func NewNFSdriver(nodeID, endpoint string) *nfsDriver {
+func loadControllerPlugin(pluginName string) (interface{}, []csi.ControllerServiceCapability_RPC_Type, error) {
+	csc := []csi.ControllerServiceCapability_RPC_Type{}
+
+	if pluginName == "" {
+		csc = append(csc, csi.ControllerServiceCapability_RPC_UNKNOWN)
+		return nil, csc, nil
+	}
+
+	plug, err := plugin.Open(pluginName)
+	if err != nil {
+		glog.Infof("Failed to load plugin: %s error: %v", pluginName, err)
+		return nil, csc, err
+	}
+
+	csPlugin, err := plug.Lookup(pluginSymbolName)
+	if err != nil {
+		glog.Infof("Failed to lookup csPlugin: %s error: %v", pluginSymbolName, err)
+		return nil, csc, err
+	}
+
+	// Check if csPlugin implements each capability and add it to implenentation
+	if _, ok := csPlugin.(ControllerPlugin); !ok {
+		glog.Infof("Plugin doesn't implement mandatory methods for controller")
+		return nil, csc, fmt.Errorf("Plugin doesn't implement mandatory methods for controller")
+	}
+
+	if _, ok := csPlugin.(CreateDeleteVolumeControllerPlugin); ok {
```

Comment:
I would consider changing the following chain to of `if` statements to a type switch:

```go
switch csPlugin.(type) {
   case CreateDeleteVolumeControllerPlugin:
       csc = append(csc, csi.ControllerServiceCapability_RPC_CREATE_DELETE_VOLUME)
   case PublishUnpublishVolumeControllerPlugin:
   ...
   default:
}
```